packages/auth/amplify_auth_cognito/lib/src/auth_plugin_impl.dart

```diff
@@ -163,4 +173,158 @@ class _NativeAmplifyAuthCognito implements NativeAuthPlugin {
       _stateMachine.addInstance(oauthParameters);
     }
   }
+
+  FutureOr<String> _getBundleId() {
+    return _bundleIdMemoizer.runOnce(() {
+      final bridge = _stateMachine.expect<NativeAuthBridge>();
+      return bridge.getBundleId();
+    });
+  }
+
+  @override
+  Future<CredentialStoreData?> fetchLegacyCredentials({
+    CognitoUserPoolConfig? userPoolConfig,
+    CognitoIdentityCredentialsProvider? identityPoolConfig,
+    CognitoOAuthConfig? hostedUiConfig,
+  }) async {
+    // TODO(Jordan-Nelson): Add credentials migration support for Android
+    if (zIsWeb || !Platform.isIOS) return null;
+    final bundleId = await _getBundleId();
+    CognitoUserPoolTokens? userPoolTokens;
+    if (userPoolConfig != null) {
+      final userPoolStorage = getUserPoolStorage(bundleId);
```

Comment:
Since credential keys are namespaced by identity pool id and app client ID, if the config is switched the old key-value pairs will be ignored, but the data isn't cleared out (like it is on iOS) if the config is switched. While developing, you can actually switch back and forth between two separate configs and remain authenticated for both.

I think it is debatable if this is what we want or not. I am not sure what the use case for switching configs in production would be and what the desired experience would be. In development, I could see either experience being desirable, but I think the production use case (if one exists) is probably more important.

This sort of ties into app uninstall use cases. Whatever API is exposed to clear keys on uninstall/re-install should delete all keys for all configs.